consent/strategy_default.go

```diff
@@ -1151,6 +1173,51 @@ func (s *DefaultStrategy) HandleOAuth2AuthorizationRequest(
 	return consentSession, f, nil
 }
 
+// HandleOAuth2DeviceAuthorizationRequest handles the device authorization flow
+func (s *DefaultStrategy) HandleOAuth2DeviceAuthorizationRequest(
+	ctx context.Context,
+	w http.ResponseWriter,
+	r *http.Request,
+) (*flow.AcceptOAuth2ConsentRequest, *flow.Flow, error) {
+	deviceVerifier := strings.TrimSpace(r.URL.Query().Get("device_verifier"))
+	loginVerifier := strings.TrimSpace(r.URL.Query().Get("login_verifier"))
+	consentVerifier := strings.TrimSpace(r.URL.Query().Get("consent_verifier"))
+
+	if deviceVerifier == "" && loginVerifier == "" && consentVerifier == "" {
+		// ok, we need to process this request and redirect to device auth endpoint
+		return nil, nil, s.requestDevice(ctx, w, r)
+	} else if deviceVerifier != "" && loginVerifier == "" && consentVerifier == "" {
+		f, err := s.verifyDevice(ctx, w, r, deviceVerifier)
+		if err != nil {
+			return nil, nil, err
+		}
+		ctx = context.WithValue(ctx, ctxKeyDeviceFlowId, f.ID)
+	}
+
+	ctx = context.WithValue(ctx, ctxKeyRequestURL, s.getDeviceVerificationPath(ctx))
```

Comment:
probably overkill to put this in the ctx, it's easily retrievable and can be called from anywhere it's needed (unless i missed something)